File: server/FitMate.Core/JsonModels/Subscriptions/UsageReservationModel.cs
using FitMate.DB.Enums;

namespace FitMate.Core.JsonModels.Subscriptions;

public class UsageReservationModel
{
    public long Id { get; set; }
    public SubscriptionFeature Feature { get; set; }
    public int Quantity { get; set; }
    public UsageReservationStatus Status { get; set; }
    public DateTime ExpiresAt { get; set; }
}

File: client/src/types/JsonModels/Subscriptions/UsageReservationModel.ts
import type { SubscriptionFeature } from "../Enums/SubscriptionFeature";
import type { UsageReservationStatus } from "../Enums/UsageReservationStatus";

export interface UsageReservationModel
{
	id: number;
	feature: SubscriptionFeature;
	quantity: number;
	status: UsageReservationStatus;
	expiresAt: string;
}
